src/internal/connector/onedrive/restore.go

```diff
@@ -669,14 +714,70 @@ func createRestoreFolders(
 	return parentFolderID, nil
 }
 
+func copyBufferWithStallCheck(dst io.Writer, src io.Reader, stallTimeout time.Duration) (int64, error) {
+	buffer := make([]byte, copyBufferSize)
```

Comment:
since these buffers are rather large it may actually be worth making a buffer pool for them. We can use sync.Pool, we just need to make sure we don't free them before we're done with them